tutorials/v7/draw_mt.cxx

```diff
@@ -87,9 +109,14 @@ void draw_mt()
    gEnv->SetValue("WebGui.HttpThrd","yes");
    gEnv->SetValue("WebGui.SenderThrds","yes");
 
-   std::thread thrd1(draw_canvas, "First canvas", RColor::kRed);
-   std::thread thrd2(draw_canvas, "Second canvas", RColor::kBlue);
-   std::thread thrd3(draw_canvas, "Third canvas", RColor::kGreen);
+   ROOT::EnableThreadSafety();
```

Comment:
Isn't this required already by "WebGui.SenderThrds" being "yes", i.e. if I modify my rootrc? In that case whoever reacts to "WebGui.SenderThrds" being "yes" must call `ROOT::EnableThreadSafety()`

Reply:
`ROOT::EnableThreadSafety()` required when running I/O from many threads. 
If simply do WebGui.SenderThrds = yes, I/O will remain in main thread.

Reply:
It is not 'just' for I/O.  Without ROOT::EnableThreadSafety(), none (in first approximation) of the ROOT code can be safely run in multiple thread.  So this is fine as long as ROOT code is only run in the main thread (but I would venture that this is a doubtful assumption).

Reply:
Yes, it is not only I/O. But in this tutorial I was facing race condition in I/O. And rare crashes somewhere else.

Reply:
So:
> whoever reacts to "WebGui.SenderThrds" being "yes" must call ROOT::EnableThreadSafety()

still holds?

Reply:
"WebGui.SenderThrds" = "yes" is introduced for WebEve. We expect here very different quality of clients connection, therefore want to avoid that slow client blocks others. But all kinds of data access from TTree or from data base or from that-ever data server will be performed in main thread. Also all Eve methods will be executed there.
Of course, functionality will be useful for all other web-based apps.